docs/reference/secret.md

```diff
@@ -224,8 +230,9 @@ Juju maintains a list of which observers are tracking each revision of each secr
 
 ```{important}
 
-When a unit gets a secret for the first time it will automatically be set to track the latest revision. A unit cannot choose to track an outdated revision, but it can in principle refuse to update to a newer one.
+Charms that create secrets should _always_ handle the `secret-remove` event. Secret revisions, even if obsolete, remain until removed by the charm. Failing to do so will cause them to accumulate indefinitely.
```

Comment:
suggestion: Rephrase to improve clarity:

```suggestion
Charms that create secrets should _always_ handle the `secret-remove` event. That is because secret revisions, even if obsolete, remain until removed by the charm; if a charm does not remove them, they accumulate indefinitely.
```